fastplotlib/layouts/_plot_area.py

```diff
@@ -696,29 +621,34 @@ def delete_graphic(self, graphic: Graphic):
             The graphic to delete
 
         """
-        # TODO: proper gc of selectors, RAM is freed for regular graphics but not selectors
-        # TODO: references to selectors must be lingering somewhere
-        # TODO: update March 2024, I think selectors are gc properly, should check
-        # get memory address
-        address = graphic._fpl_address
-
         if graphic not in self:
             raise KeyError(f"Graphic not found in plot area: {graphic}")
 
-        # check which type it is
-        for l in [self._graphics, self._selectors, self._legends]:
-            if address in l:
-                l.remove(address)
-                break
+        if isinstance(graphic, BaseSelector):
+            self._selectors.remove(graphic)
+        elif isinstance(graphic, Legend):
+            self._legends.remove(graphic)
+
+        elif isinstance(graphic, Graphic):
+            self._graphics.remove(graphic)
 
         # remove from scene if necessary
         if graphic.world_object in self.scene.children:
             self.scene.remove(graphic.world_object)
 
         # cleanup
-        graphic._fpl_cleanup()
-
-        REFERENCES.remove(address)
+        graphic._fpl_prepare_del()
+
+        if IS_IPYTHON:
+            # remove any references that ipython might have made
+            ip = get_ipython()
+            # get all references in ipython namespace
+            references = list(ip.user_ns.keys())
+            for ref in references:
+                if ip.user_ns[ref] is graphic:
+                    # we found the graphic, remove from ipython
+                    ip.del_var(ref)
+                    break
```

Comment:
How likely is it that the object is in `user_ns`? If it's only occasionally, can wrap this in:
```py
if graphic in ip.user_ns.values():
    ...
```

I guess it's possible that the graphic is present more than once, under a different name?

Alt solution that might be faster because it avoids copying the keys and looking up the values:
```py
            keys_to_delete = []
            for key, ob in ip.user_ns.items():
                if ob is graphic:
                    keys_to_delete.append(key)
            for key in keys_to_delete:
                    ip.del_var(key)
```

Reply:
> How likely is it that the object is in `user_ns`? If it's only occasionally, can wrap this in:
> 
> ```python
> if graphic in ip.user_ns.values():
>     ...
> ```

`in` operators don't work if any item of the iterable is a numpy array, for example:

```python
a = np.random.rand(10)
l = [a, "bah", 123]

# will raise a ValueError
123 in l
```

I've resorted to this, couldn't think of a more performance solution:

```python
        if IS_IPYTHON:
            # remove any references that ipython might have made
            ip = get_ipython()

            # check both namespaces
            for namespace in [ip.user_ns, ip.user_ns_hidden]:
                # find the reference
                for ref, obj in namespace.items():
                    if graphic is obj:
                        # we found the reference, remove from ipython
                        ip.del_var(ref)
                        break
```

Reply:
No problems with deleting items from an object being iterated over? I guess it depends on IPython internals whether that'd be a problem.